packages/aws-cdk-lib/aws-elasticloadbalancingv2/lib/alb/application-listener.ts

```diff
@@ -96,6 +97,62 @@ export interface BaseApplicationListenerProps {
    * @default true
    */
   readonly open?: boolean;
+
+  /**
+   * The mutual authentication configuration information
+   *
+   * @default - No mutual authentication configuration
+   *
+   * @see https://docs.aws.amazon.com/elasticloadbalancing/latest/application/mutual-authentication.html
+   */
+  readonly mutualAuthentication?: MutualAuthentication;
+}
+
+/**
+ * The mutual authentication configuration information
+ *
+ */
+export interface MutualAuthentication {
+  /**
+   * Indicates whether expired client certificates are ignored
+   *
+   * @default false
+   */
+  readonly ignoreClientCertificateExpiry?: boolean;
+
+  /**
+   * The client certificate handling method
+   *
+   * @default MutualAuthenticationMode.OFF
+   */
+  readonly mutualAuthenticationMode?: MutualAuthenticationMode;
+
+  /**
+   * The trust store
+   *
+   * @default - no trust store
+   */
+  readonly trustStore?: ITrustStore;
```

Comment:
In the constructor (lines 248-261) we're checking whether `trustStore` and `mutualAuthenticationMode` are being set separately, and throwing an error in cases where they are either required or not required. (i.e. if mode is `VERIFY`, then `trustStore` must be set) Can we document those relations in these docstrings?

Reply:
I added an explanation. Considering the possibility that modes other than "verify" might be added in the future, I've included a note stating that this cannot be used when the mode is "off" or "passthrough".